nav2_planner/src/planner_server.cpp

```diff
@@ -510,7 +517,72 @@ PlannerServer::publishPlan(const nav_msgs::msg::Path & path)
   }
 }
 
-}  // namespace nav2_planner
+void PlannerServer::isPathValid(
+  const std::shared_ptr<nav2_msgs::srv::IsPathValid::Request> request,
+  std::shared_ptr<nav2_msgs::srv::IsPathValid::Response> response)
+{
+  unsigned int mx = 0;
+  unsigned int my = 0;
+
+  response->is_valid = true;
+
+  if (request->path.poses.empty() ) {
+    response->is_valid = false;
+  } else {
+    geometry_msgs::msg::PoseStamped current_pose;
+    unsigned int closest_point_index = 0;
+    if (costmap_ros_->getRobotPose(current_pose)) {
+      float closest_distance = std::numeric_limits<float>::max();
+      float current_distance = std::numeric_limits<float>::max();
+      float prev_distance = std::numeric_limits<float>::max();
+      unsigned int increasing_distance_counter = 0;
+      const unsigned int MAX_INCREASE = 3;
+      geometry_msgs::msg::Point current_point;
+      current_point = current_pose.pose.position;
+      for (unsigned int i = 0; i < request->path.poses.size(); ++i) {
+        geometry_msgs::msg::Point path_point;
+        path_point.x = request->path.poses[i].pose.position.x;
+        path_point.y = request->path.poses[i].pose.position.y;
+        path_point.z = request->path.poses[i].pose.position.z;
+
+        current_distance = nav2_util::geometry_utils::euclidean_distance(
+          current_point,
+          path_point);
+
+        if (current_distance < closest_distance) {
+          closest_point_index = i;
+          closest_distance = current_distance;
+        }
+
+        if (current_distance > prev_distance) {
```

Comment:
Why don't you revert this logic and I'll take care of it as part of https://github.com/ros-planning/navigation2/issues/2622 when we have a solution I like for that problem to apply to all of the places at once. Sorry about that, just do the normal marching through to find the closest point.